src/config/workload.rs

```diff
@@ -468,3 +466,57 @@ impl Verb {
         matches!(self, Self::ListPushBack | Self::ListPushFront)
     }
 }
+
+#[derive(Clone, Deserialize)]
+pub struct Ratelimit {
+    #[serde(default)]
+    ratelimit: u64,
+
+    #[serde(default)]
+    ratelimit_end: Option<u64>,
+
+    #[serde(default)]
+    step_size: Option<u64>,
+
+    #[serde(default)]
+    step_interval: Option<u64>,
+}
+
+impl Ratelimit {
+    pub fn start(&self) -> Option<NonZeroU64> {
+        NonZeroU64::new(self.ratelimit)
+    }
+
+    pub fn end(&self) -> Option<u64> {
+        self.ratelimit_end
+    }
+
+    pub fn step(&self) -> Option<u64> {
+        self.step_size
+    }
+
+    pub fn interval(&self) -> Option<Duration> {
+        self.step_interval.map(Duration::from_secs)
+    }
+
+    pub fn is_dynamic(&self) -> bool {
+        self.ratelimit_end.is_some() || self.step_size.is_some() || self.step_interval.is_some()
+    }
+
+    pub fn validate_dynamic(&self) -> bool {
```

Comment:
maybe we should just have a public validate function? Would seem to make more sense to always validate rather than only when its dynamic.

Reply:
Alright, done.